### ISSUE TITLE
snap to compact on re-sizing to a sufficiently narrow width

### ISSUE THREAD

**Issue by akhmorgan** [ScottLogic/bitflux-openfin#352]
When re-sizing the main window, should the width be reduced to a sufficiently small amount, the app will snap to the compact view.

**Comment by mbssantos** [comment]
Got a few UX questions about this one @rmarcalo 

For reference:
Minimum expanded window size is 918 x 510
Collapsed window size is 230 x 500

### Expanded to collapsed
Assuming we don't want to snap from 918 to 230, should we be fully responsive?
For clarity, I'm thinking something along the lines of  [bootstrap](http://getbootstrap.com/)'s website.

### Collapsed to expanded
I'm assuming we also want this although it's not mentioned?

### BitFlux Graph
when the graph area is under 600, content starts overlapping with itself and with the favorites bar.
Should we hide the graph area and just have a big favorites bar?

**Comment by owennw** [comment]
Collapsed to expanded: collapsed view cannot be resized.

**Comment by rmarcalo** [comment]
This was suggested a while ago, but I eventually backed away from it for a few reasons. One of them is that there is a great jump between the minimum acceptable size for the chart, and the width of the compact view, as you mentioned. There isn't really an in-between solution that keeps functionality and usefulness, defeating the purpose of full responsiveness.  Since having the chart unusable or unreadable is not desirable, I thought it was best to treat the compact view as a toggle, rather than an effect of manual resize.

**Comment by mbssantos** [comment]
Snapping it is then!
Do we want any "break" when the user reaches the expanded window's minimum width or just snap immediately?
I'm thinking If a user wants the smallest possible _expanded_ window, he'll have a hard time getting it..

Also, shouldn't we have the opposite functionality for collapsed -> expanded? 

Apologies if you've been through these points, is there anywhere I can get up to date on what's been proposed?

**Comment by rmarcalo** [comment]
I was suggesting not having any snapping even. You can resize until the minimum expanded window and if you want the compact size you use the toggle. 

If we go with the snapping I would expect a reasonable 50px (or more) break in which the user is making the window narrower but it doesn't budge, snapping to compact if it is exceeded.

**Comment by owennw** [comment]
Yes I interpreted @rmarcalo's earlier comment as a "close this issue, don't do". Is that correct?

**Comment by rmarcalo** [comment]
It's a debatable feature, which we could perhaps discuss in a future call.

**Comment by owennw** [comment]
Don't know if this should change milestone, @ColinEberhardt ?

**Comment by ColinEberhardt** [comment]
We'll work out the next milestone after this release :-)

**Comment by bjedrzejewski** [comment]
I side with @rmarcalo I think this is the kind of feature that many people will not like, so I don't think it justifies the development.

**Comment by ColinEberhardt** [comment]
I'd certainly like to debate this feature further, rather than close it, but the priority is relatively low. We'll keep as a candidate for the next release.
